stackslib/src/chainstate/nakamoto/coordinator/mod.rs

```diff
@@ -63,9 +65,21 @@ impl OnChainRewardSetProvider {
             .block_height_to_reward_cycle(cycle_start_burn_height)
             .expect("FATAL: no reward cycle for burn height");
 
-        let registered_addrs =
+        let mut registered_addrs =
             chainstate.get_reward_addresses_in_cycle(burnchain, sortdb, cycle, block_id)?;
 
+        // TODO (pox-4-workstream): the pox-4 contract must be able to return signing keys
```

Comment:
Is there an issue for this?  Because this seems to be pretty high-priority!

Reply:
There's a lot of ongoing discussion of this in the open .signers PR, I'm not sure that it has its own issue yet, but its definitely top of mind there.

Reply:
Opened an issue: https://github.com/stacks-network/stacks-core/issues/4290